### PR TITLE
Rollup of 20 pull requests

### PR DESCRIPTION
**Pull request by steveklabnik**
- Successful merges: #26778, #27084, #27127, #27131, #27140, #27146, #27147, #27151, #27154, #27155, #27156, #27157, #27158, #27164, #27166, #27167, #27170, #27175, #27201, #27202
- Failed merges:

### CODE REVIEW

**Comment by barosl** [comment]
What a huge priority number!

File: src/libstd/io/buffered.rs [hunk 1]
```diff
@@ -449,12 +698,45 @@ impl<W: Write> LineWriter<W> {
     ///
     /// Caution must be taken when calling methods on the mutable reference
     /// returned as extra writes could corrupt the output stream.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fs::File;
+    /// use std::io::prelude::*;
```
Comment (apasel422) [inline src/libstd/io/buffered.rs]: It appears that this import was unused and caused the test failure.

**Comment by steveklabnik** [comment]
ugh :( changing examples at the last minute.... i thought i had fixed this.

**Pull request closed without merging**